gateway/oauth/src/main/kotlin/org/doorip/gateway/oauth/apple/ApplePublicKeyGenerator.kt

```
@@ -0,0 +1,36 @@
+
+        val alg = header["alg"] ?: throw UnauthenticatedException
+        val kid = header["kid"] ?: throw UnauthenticatedException
+        val key = applePublicKeys[alg, kid] ?: throw UnauthenticatedException
+
+        return RSAPublicKey(key.n, key.e, key.kty)
+    }
+
+    private fun getApplePublicKeys(): ApplePublicKeys {
+        try {
+            return httpClient.getApplePublicKeys()
+        } catch (e: FeignException) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)